plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java

```diff
@@ -560,15 +560,40 @@ private void stageFlexYamlTemplate(
       String currentTemplateName,
       String imagePath,
       File metadataFile,
-      String yamlTemplateName,
+      String yamlTemplateFile,
       String templatePath)
       throws IOException, InterruptedException, TemplateException {
 
-    // TODO(polber) Use basePythonContainerImage once plugin can parse metadata from YAML Templates
-    String containerImage = "gcr.io/" + projectId + "/beam-yaml/yaml-template-base:latest";
+    String beamVersion = project.getProperties().getProperty("beam.version");
+    String yamlTemplateName = yamlTemplateFile.replace(".yaml", "");
+    List<String> otherFiles = new ArrayList<>();
+    String filesToCopy = definition.getTemplateAnnotation().filesToCopy();
+    if (!Strings.isNullOrEmpty(filesToCopy)) {
+      otherFiles.addAll(List.of(filesToCopy.split(",")));
+    }
+    if (!Strings.isNullOrEmpty(yamlTemplateFile)) {
+      otherFiles.add(yamlTemplateFile);
+    }
     YamlDockerfileGenerator.generateDockerfile(
-        containerImage, yamlTemplateName, outputClassesDirectory);
-    stageYamlUsingDockerfile(imagePath, yamlTemplateName + "/Dockerfile");
+        basePythonContainerImage,
+        baseContainerImage,
+        beamVersion,
+        yamlTemplateName,
+        otherFiles,
+        outputClassesDirectory);
+
+    boolean useRootDirectory = true;
+    if (new File(outputClassesDirectory.getPath() + "/" + yamlTemplateName + "/main.py").exists()) {
```

Comment:
We're duplicating this logic in a couple places - instead of just passing in a boolean, could we pass a string like entrypointDirectory or something that resolves to the relative path here? Then all we need to worry about below is `/`s which can be handled with libraries.

Reply:
The logic isn't exactly duplicated. There are 2 different directories we are tracking - the build directory within the plugin, and the build directory on cloud build. True YAML templates are built with a slightly different structure in both cases from the YamlTemplate.

This is step 1 in an effort to eventually migrate the YamlTemplate to the yaml module so that all yaml-related features are consolidated in one place.